File: grpc/examples/python/get-accounts.py
#!/usr/bin/python3

"""
Script language: Python3

Talks to:
- Vega node (gRPC)

Apps/Libraries:
- Vega-API-client (https://pypi.org/project/Vega-API-client/)

Responses:
- response-examples.txt
"""

# Note: this file uses smart-tags in comments to section parts of the code to
# show them as snippets in our documentation. They are not necessary to be
# included when creating your own custom code.
#
# Example of smart-tags:
#  __something:
# some code here
# :something__

import helpers

# Vega wallet interaction helper, see login.py for detail
from login import token, pubkey

# __import_client:
import vegaapiclient as vac
node_url_grpc = helpers.get_from_env("NODE_URL_GRPC")
data_client = vac.VegaTradingDataClient(node_url_grpc)
# :import_client__

print(token)

###############################################################################
#                        M A R K E T   A C C O U N T S                        #
###############################################################################

# Request a list of markets and select the first one
req = vac.api.trading.MarketsRequest()
markets = data_client.Markets(req).markets
marketID = markets[0].id

assert marketID != ""
print(f"Market found: {marketID}")

# __get_accounts_by_market:
# Request a list of accounts for a market on a Vega network
request = vac.api.trading.MarketAccountsRequest(market_id=marketID)
market_accounts = data_client.MarketAccounts(request)
print("Market accounts:\n{}".format(market_accounts))
# :get_accounts_by_market__

###############################################################################
#                         P A R T Y   A C C O U N T S                         #
###############################################################################

# __get_accounts_by_party:
# Request a list of accounts for a party (pubkey) on a Vega network
request = vac.api.trading.PartyAccountsRequest(party_id=pubkey)
party_accounts = data_client.PartyAccounts(request)
print("Party accounts:\n{}".format(party_accounts))
# :get_accounts_by_party__

###############################################################################
#                        P A R T Y   P O S I T I O N S                        #
###############################################################################

# __get_positions_by_party:
# Request a list of positions for a party (pubkey) on a Vega network
request = vac.api.trading.PositionsByPartyRequest(party_id=pubkey)
party_positions = data_client.PositionsByParty(request)
print("Party positions:\n{}".format(party_positions))
# :get_positions_by_party__

File: grpc/examples/go/get-accounts.go
package main

import (
	"fmt"

	"github.com/vegaprotocol/api/grpc/clients/go/generated/code.vegaprotocol.io/vega/proto/api"
	"github.com/vegaprotocol/api/grpc/examples/go/helpers"

	"golang.org/x/net/context"
	"google.golang.org/grpc"
)

func main() {
	// Helpers include logic to authenticate with a Vega wallet service
	// including token storage/public key operations/signing of commands
	// visit ./helpers/wallet.go for more detail
	pubkey := helpers.GetPubKey()

	nodeURLGrpc := helpers.GetFromEnv("NODE_URL_GRPC")
	conn, err := grpc.Dial(nodeURLGrpc, grpc.WithInsecure())
	if err != nil {
		panic(err)
	}
	defer conn.Close()

	dataClient := api.NewTradingDataServiceClient(conn)

	// Get market
	marketRequest := api.MarketsRequest{}
	markets, err := dataClient.Markets(context.Background(), &marketRequest)
	if err != nil {
		panic(err)
	}
	marketID := markets.Markets[0].Id

	// __get_accounts_by_market:
	// Request a list of accounts for a market on a Vega network
	accountsReq := api.MarketAccountsRequest{MarketId: marketID}
	acconutsResp, err := dataClient.MarketAccounts(context.Background(), &accountsReq)
	if err != nil {
		panic(err)
	}
	fmt.Printf("Market accounts: %v\n", acconutsResp)
	// :get_accounts_by_market__

	// __get_accounts_by_party:
	// Request a list of accounts for a party (pubkey) on a Vega network
	partyReq := api.PartyAccountsRequest{PartyId: pubkey}
	partyResp, err := dataClient.PartyAccounts(context.Background(), &partyReq)
	if err != nil {
		panic(err)
	}
	fmt.Printf("Party accounts: %v\n", partyResp)
	// :get_accounts_by_party__

	// __get_positions_by_party:
	// Request a list of positions for a party (pubkey) on a Vega network
	partyPosReq := api.PositionsByPartyRequest{PartyId: pubkey}
	partyPosResp, err := dataClient.PositionsByParty(context.Background(), &partyPosReq)
	if err != nil {
		panic(err)
	}
	fmt.Printf("Party positions: %v\n", partyPosResp)
	// :get_positions_by_party__
}
